os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -415,12 +415,13 @@ static void up_dumpstate(struct tcb_s *fault_tcb, uint32_t asserted_location)
 
 static void arm_assert(void)
 {
+#ifdef CONFIG_BOARD_ASSERT_AUTORESET
+	boardctl(BOARDIOC_RESET, EXIT_SUCCESS);
+#else
+#ifndef CONFIG_BOARD_ASSERT_SYSTEM_HALT
 	/* Are we in an interrupt handler or the idle task? */
 
-	if (CURRENT_REGS || (this_task())->flink == NULL)
-	{
-#if CONFIG_BOARD_RESET_ON_ASSERT >= 1
-		board_reset(CONFIG_BOARD_ASSERT_RESET_VALUE);
+	if (CURRENT_REGS || (this_task())->flink == NULL) {
```

Comment:
How about changing the condition for idle task to `(this_task())->pid == 0`?

Reply:
Sure, I changed,

Reply:
according to the SMP design that is going to be added in https://github.com/Samsung/TizenRT/pull/5958, there will be an idle thread for each cpu. Therefore only checking ```pid == 0``` is not enough in the case of SMP. Incase that PR merges before this, this change needs to be reverted.

Reply:
I understand, thank you, changed,